test/teststl.cpp

```diff
@@ -4006,6 +4012,22 @@ class TestStl : public TestFixture {
               "    }\n"
               "}\n");
         ASSERT_EQUALS("", errout.str());
+
+        check("void func(const std::string& b) {\n"
+              "    std::string a = b.c_str();\n"
+              "}");
+        ASSERT_EQUALS("[test.cpp:2]: (performance) Assigning the result of c_str() to a variable of type std::string is slow and redundant.\n", errout.str());
```

Comment:
hmm let's discuss the warning message a little. It's not the assignment that is redundant. But the `c_str()` call. How about:

    Redundant .c_str() call in assignment.

Or do you have a better suggestion?